lib/api/core/auth/passportWrapper.js

```diff
@@ -0,0 +1,35 @@
+var
+  q = require('q'),
+  passport = require('passport');
+
+module.exports = function PassportWrapper (kuzzle) {
+
+  this.authenticate = function(request, strategy) {
+    var
+      deferred = q.defer();
+
+    try {
+      if (!passport._strategy(strategy)) {
+        return Promise.reject(new Error('Unknown authentication strategy "' + strategy + '"'));
+      }
+      passport.authenticate(strategy, function(err, user, info) {
+        kuzzle.pluginsManager.trigger('log:silly', 'Authenticate Info : ' + info);
+        if (err !== null) {
+          deferred.reject(err);
+        }
+        else if (user === false) {
+          deferred.reject(new Error(info.message));
+        }
+        else {
+          deferred.resolve(user);
+        }
+      })(request);
+    } catch (err) {
+      deferred.reject(new Error('Error authenticating user'));
```

Comment:
hmm for the line 21, I agree to set a ForbiddenError.
Bug within the catch, maybe an InternalError is more appropriate...